pkg/ui/progressbar.go

```diff
@@ -29,52 +29,58 @@ const (
 // It is used to show the progress of some running task (or multiple).
 // Example (Infinite Progress without a value):
 //
-//	var pBar ProgressBar = ui.DefaultUi().NewProgressBar()
+//	pBar := ui.DefaultUi().NewProgressBar("CLI Downloader")
 //	defer pBar.Clear()
-//	pBar.SetTitle("Downloading...")
+//	pBar.SetMessage("Downloading...")
 //	_ = pBar.UpdateProgress(ui.InfiniteProgress)
 //
 // Example (with a value):
 //
-//	var pBar ProgressBar = ui.DefaultUi().NewProgressBar()
+//	pBar := ui.DefaultUi().NewProgressBar("CLI Downloader")
 //	defer pBar.Clear()
-//	pBar.SetTitle("Downloading...")
+//	pBar.SetMessage("Downloading...")
 //	for i := 0; i <= 50; i++ {
 //	    pBar.UpdateProgress(float64(i) / 100.0)
 //	    time.Sleep(time.Millisecond * 50)
 //	}
 //
-//	pBar.SetTitle("Installing...")
+//	pBar.SetMessage("Installing...")
 //	for i := 50; i <= 100; i++ {
 //	    pBar.UpdateProgress(float64(i) / 100.0)
 //	    time.Sleep(time.Millisecond * 50)
 //	}
 //
-// The title can be changed in the middle of the progress bar.
+// The message can be changed in the middle of the progress bar.
+// In some implementations no progress bar will be shown until the first UpdateProgress is called.
 type ProgressBar interface {
 	// UpdateProgress updates the state of the progress bar.
 	// The argument `progress` should be a float64 between 0 and 1,
 	// where 0 represents 0% completion, and 1 represents 100% completion.
 	// Returns an error if the update operation fails.
 	UpdateProgress(progress float64) error
 
-	// SetTitle sets the title of the progress bar, which is displayed next to the bar.
-	// The title provides context or description for the operation that is being tracked.
-	SetTitle(title string)
+	// SetMessage sets the message of the progress bar, which is displayed next to the bar.
+	// The message is displayed alongside the title passed in by `NewProgressBar(title string)`.
+	// The message provides context or description for the operation that is being tracked.
+	// Set to "" to clear the message.
+	SetMessage(message string)
```

Comment:
Question: @rrama did you integrate these changes in the CLI? I assume given that this PR changes interfaces, it'll require changes to existing extensions. Did you consider doing this without a breaking change?